src/main/java/com/powsybl/balances_adjustment/balance_computation/BalanceComputationParameters.java

```diff
@@ -6,59 +6,116 @@
  */
 package com.powsybl.balances_adjustment.balance_computation;
 
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.powsybl.commons.config.PlatformConfig;
+import com.powsybl.commons.extensions.*;
 import com.powsybl.loadflow.LoadFlowParameters;
 
+import java.util.Objects;
+
 /**
  * parameters for balance computation.
+ *
  * @author Ameni Walha {@literal <ameni.walha at rte-france.com>}
+ * @author Mohamed Ben Rejeb {@literal <mohamed.benrejeb at rte-france.com>}
  */
-public class BalanceComputationParameters {
+public class BalanceComputationParameters extends AbstractExtendable<BalanceComputationParameters> {
 
-    private final LoadFlowParameters loadFlowParameters;
+    public static final double DEFAULT_THRESHOLD_NET_POSITION = 1;
+    public static final int DEFAULT_MAX_NUMBER_ITERATIONS = 5;
 
     /**
      * Threshold for comparing net positions (given in MW).
      * Under this threshold, the network area is balanced
      */
-    private final double thresholdNetPosition;
+    private double thresholdNetPosition;
 
     /**
      * Maximum iteration number for balances adjustment
      */
-    private final int maxNumberIterations;
-
-    private static final double DEFAULT_THRESHOLD_NETPOSITION = 1;
-
-    private static final int DEFAULT_MAX_NUMBER_ITERATIONS = 5;
-
+    private int maxNumberIterations;
 
     /**
      * Constructor with default parameters
      */
     public BalanceComputationParameters() {
-        this(DEFAULT_THRESHOLD_NETPOSITION, DEFAULT_MAX_NUMBER_ITERATIONS, LoadFlowParameters.load());
+        this(DEFAULT_THRESHOLD_NET_POSITION, DEFAULT_MAX_NUMBER_ITERATIONS);
     }
 
     /**
      * Constructor with given parameters
      * @param threshold Threshold for comparing net positions (given in MW)
-     * @param max Maximum iteration number for balances adjustment
+     * @param maxNumberIterations Maximum iteration number for balances adjustment
      */
-    public BalanceComputationParameters(double threshold, int max, LoadFlowParameters loadFlowParameters) {
+    public BalanceComputationParameters(double threshold, int maxNumberIterations) {
         this.thresholdNetPosition = threshold;
-        this.maxNumberIterations = max;
-        this.loadFlowParameters = loadFlowParameters;
+        this.maxNumberIterations = maxNumberIterations;
```

Comment:
You should assert that both thresholdNetPosition and maxNumberIterations are positive. To do that, add static private method like this:
```
private static final double checkThresholdNetPosition(double threshold) {
    if (threshold < 0) {
        throw new IllegalArgumentException("Threshold must be positive");
    }
    return threshold;
}
private static final int checkMaxNumberIterations(int maxNumberIterations) {
    if (maxNumberIterations < 0) {
        throw new IllegalArgumentException("The maximum number of iterations must be positive");
    }
    return maxNumberIterations;
}
```